src/core/error/common_error.rs

```diff
@@ -492,6 +492,9 @@ pub enum CommonError {
 
     #[error("Invalid RadixConnectPurpose, bad value: {bad_value}")]
     InvalidRadixConnectPurpose { bad_value: String } = 10137,
+
+    #[error("Index out of bounds: {index} >= #{count}")]
+    IndexOutOfBounds { index: u64, count: u64 } = 10138,
```

Comment:
Hmm, I think this error is not too helpful for us debugging, we should use a more specific one:
`TXGuaranteeIndexOutOfBounds { ... }` (only used for this scenario).

At least until we have come up with some genius "error system", because now we don't have any "context" and when left as `IndexOutOfBounds` it makes sense to use it elsewhere for possible OOB errors, and then it will make debugging harder, when we are given an Error Code by user. See how detailed I've been for other errors:
```rust
    InvalidEd25519PrivateKeyFromBytes { bad_value: BagOfBytes } = 10001,

    #[error("Failed to create Ed25519 Private key from String {bad_value}.")]
    InvalidEd25519PrivateKeyFromString { bad_value: String } = 10002,

    #[error(
        "Failed to create Secp256k1 Private key from bytes {bad_value:?}."
    )]
    InvalidSecp256k1PrivateKeyFromBytes { bad_value: BagOfBytes } = 10003,

    #[error(
        "Failed to create Secp256k1 Private key from String {bad_value:?}."
    )]
    InvalidSecp256k1PrivateKeyFromString { bad_value: String } = 10004,
```

These four could have been 1, `InvalidPrivateKey`, but now we get much more details, we know which curve and from which input.

Reply:
Sure, I really don't have an opinion on the name of the error, it has been changed now